packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts

```diff
@@ -134,115 +225,125 @@ export class EditorSettingsSidebarComponent {
-	 *
-	 * For instance, if the current time is in the 'am' and the article is
-	 * scheduled for 'pm', the act of clicking on the 'pm' button dismisses the
-	 * scheduler.
+	 * Closes the Schedule picker.
 	 */
 	async closeSchedule(): Promise< void > {
-		const expanded = await this.frame.getAttribute( selectors.scheduleButton, 'aria-expanded' );
-		if ( expanded !== 'false' ) {
-			await this.frame.click( selectors.scheduleButton );
+		// Under certain circumstances, the scheduler is auto-dismissed at the end
+		// of interaction and thus the closure is not required.
+		// For instance, if the current time is in the 'am' and the article is
+		// scheduled for 'pm', the act of clicking on the 'pm' button dismisses the
+		// scheduler.
+		if ( ! ( await this.targetIsOpen( selectors.scheduleButton ) ) ) {
+			return;
 		}
-		await this.frame.waitForSelector( `${ selectors.scheduleButton }[aria-expanded="false"]` );
+
+		const buttonLocator = this.frameLocator.locator( selectors.scheduleButton );
+		await buttonLocator.click();
 	}
 
 	/**
 	 * Schedules the page/post.
 	 *
 	 * @param {Schedule} date Date of the article to be scheduled.
 	 */
-	async schedule( date: Schedule ): Promise< void > {
+	async setScheduleDetails( date: Schedule ): Promise< void > {
 		let key: keyof Schedule;
 
 		for ( key in date ) {
 			if ( key === 'meridian' ) {
 				// am/pm is a button.
-				await this.frame.click( selectors.scheduleMeridianButton( date[ key ] ) );
+				const meridianButtonLocator = this.frameLocator.locator(
+					selectors.scheduleMeridianButton( date[ key ] )
+				);
+				await meridianButtonLocator.click();
 				continue;
 			}
 			if ( key === 'month' ) {
 				// For month numbers less than 10, pad the digit to be
 				// 2 digits as required by the select.
-				await this.frame.selectOption(
-					'select[name="month"]',
-					date[ key ].toString().padStart( 2, '0' )
-				);
+				const monthSelectLocator = this.frameLocator.locator( selectors.scheduleMonthSelect );
+				await monthSelectLocator.selectOption( date[ key ].toString().padStart( 2, '0' ) );
 				continue;
 			}
 
 			// Regular input fields.
-			const targetSelector = selectors.scheduleInput( key );
-			await this.frame.fill( targetSelector, date[ key ].toString() );
+			const inputLocator = this.frameLocator.locator( selectors.scheduleInput( key ) );
+			await inputLocator.fill( date[ key ].toString() );
 		}
 	}
 
-	/**
-	 * Resets previously set schedule.
-	 */
-	async resetSchedule(): Promise< void > {
-		await this.frame.click( selectors.scheduleResetButton );
-	}
+	/* Revisions */
 
 	/**
 	 * Clicks on the Revisions section in the sidebar to show a revisions modal.
 	 */
 	async showRevisions(): Promise< void > {
-		await this.frame.click( selectors.revisionsToggle );
+		const locator = this.frameLocator.locator(
+			`${ selectors.sectionHeader }:has-text("Revisions")`
+		);
+		await Promise.all( [ this.page.waitForResponse( /.*diffs\?.*/ ), locator.click() ] );
```

Comment:
Do we need to wait on the web request? I assume anything that is using the revisions popup will use auto-waiting before interacting with anything on the popup, so this might just add an unnecessary wait that could flake at some point in the future.

Reply:
I felt like it was acceptable for this method to perform a guarantee, but I can remove it.

Reply:
Removed the Promise.all and response check.